### DISCUSSION TITLE
Python 3.11 _frame issue

### DISCUSSION THREAD

**Discussion by SeaOtocinclus** [pybind/pybind11#4333] [Q&A] [answered]
Hello, I'm trying to use pybind11 with python 3.11 and I got this error
```
pybind11/include/pybind11/detail/type_caster_base.h:448:36: error: member access into incomplete type 'PyFrameObject' (aka '_frame')
                "  " + handle(frame->f_code->co_filename).cast<std::string>() +
                                   ^
/Library/Frameworks/Python.framework/Versions/3.11/include/python3.11/pytypedefs.h:22:16: note: forward declaration of '_frame'
typedef struct _frame PyFrameObject;
```
Can you advise me how to handle this, or how we can update pybind11 regarding this.

Here is what I found about Python 3.11 update:

https://docs.python.org/3.11/whatsnew/3.11.html

The PyFrameObject structure members have been removed from the public C API.

While the documentation notes that the PyFrameObject fields are subject to change at any time, they have been stable for a long time and were used in several popular extensions.

In Python 3.11, the frame struct was reorganized to allow performance optimizations. Some fields were removed entirely, as they were details of the old implementation.

**Comment by Skylion007** [comment, marked as answer]
This has been fixed in the latest version of pybind11. Make sure you have the latest version installed.

**Reply by SeaOtocinclus** [reply]
Thank you @Skylion007 for your fast answer!

**Reply by matplo** [reply]
What version exactly?

**Reply by matplo** [reply]
Nevermind. Looks like 2.10.4 does the trick. Thanks!

**Reply by JonOuyang** [reply]
I'm having a similar problem with visual code giving me errors "use of undefined type '_frame', I have the latest 2.10.4 pybind11 but pyopenpose still won't build.

**Reply by subhacom** [reply]
Same issue with pybind11 v 2.11.1 and Python 3.11.6 on MacOSX Ventura (M2).

**Reply by ykchong45** [reply]
This solves the problem for me 🎉! I'm using 
```
Python v3.11.5
pybind11 v2.12.0
```

**Reply by peekxc** [reply]
... Still having this problem on MacOS Sonoma (Intel), Python v3.11.10 and any of the listed pybind11 versions

**Comment by nethtrick** [comment]
I'm getting this error with PyBind11 2.11.1 and Python 3.11.5 (both the latest I have access to at my org). I'm using Windows 11 Pro 22H2 and Visual Studio 2022 Community (14.37.32822).
